src/product/product.service.ts

```diff
@@ -28,6 +28,13 @@ export class ProductService implements IProductService {
   async createProduct(product: CreateProductDTO): Promise<ProductResponseDto> {
     const newProduct = await this.productRepository.save(product);
     return ProductResponseDto.mapToResponse(newProduct);
+  }
 
+  deleteProduct(id: string): Promise<ProductResponseDto>  {
+    this.productRepository.update(id, {isActive:false});
+    return ProductResponseDto.response(this.productRepository.findOne({
```

Comment:
There's not need return the product because front team don't need one entity removed